backend/daemon/daemon.go

```diff
@@ -193,11 +195,25 @@ func (a *App) Wait() error {
 
 func initRepo(cfg config.Config, device crypto.PrivKey) (r *ondisk.OnDisk, err error) {
 	log := logging.New("mintter/repo", "debug")
-
-	if device == nil {
+	var deviceKey crypto.PrivKey = device
+	if deviceKey == nil && cfg.Identity.DeviceKeyPath != "" {
+		if _, err := os.Stat(cfg.Identity.DeviceKeyPath); err == nil {
+			bytes, err := ioutil.ReadFile(cfg.Identity.DeviceKeyPath)
+			if err != nil {
+				return nil, err
+			}
+			deviceKey, err = crypto.UnmarshalPrivateKey(bytes)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			return nil, err
+		}
+	}
```

Comment:
This function already accepts a device private key as a parameter. It makes it slightly more complicated than necessary having this logic inside this function. Consider retrieving the private key from the file in the outser scope, and pass the parsed device key to this function later.